menhir-recover/graph.ml

```diff
@@ -0,0 +1,77 @@
+open MenhirSdk.Cmly_api
+
+module Make (G : GRAMMAR):
+sig
+  val print : string -> unit
+end = struct
+  open G
+  open Format
+
+  let print_table ?(header = "") ?(options = "") ppf table =
+    fprintf ppf "<table>\n<th>%s</th>\n" header;
+    List.iter (fun row -> 
+      fprintf ppf "<tr>";
+      List.iter (fun cell ->
+        fprintf ppf "<td%s>" options;
+        cell ();
+        fprintf ppf "</td\n>";
+      ) row;
+      fprintf ppf "</tr>\n";
+    ) table;
+    fprintf ppf "</table>\n"
+  
+  let print_LR0 lr ppf =
+    let table = [
+      [fun () -> fprintf ppf "Incoming"];
+      [fun () -> 
+      (match Lr0.incoming (Lr1.lr0 lr) with
+        | None -> fprintf ppf "None"
+        | Some symbol -> Print.symbol ppf symbol)];
+      [fun () -> fprintf ppf "Items"]
+      ] @ 
+      List.map (fun item ->
+        [fun () -> Print.item ppf item]
+      ) @@ Lr0.items @@ Lr1.lr0 lr in
+    print_table ppf table ~header:"LR0"
+
+  let print_transitions lr ppf =
```

Comment:
I think `ppf` should be the first argument, because it can be important for partial applying. Same for `print_LR0` and `print_reductions`

Reply:
Initially, I made it to apply cell to ppf, but forgot about it. Does it have more sense now?

Reply:
Yeah, Now it's good.